### PR TITLE
WIP: Cleanup Vagrant and make it use userdata.txt

### PR DESCRIPTION
**Pull request by saneax**
1. smoothing out Vagrantfile to run "puppet apply --tags packages first"
2. Adding a default sources.list, this does the initial package pull faster. (WIP)
3. adding provision to move to a configurable mirror

### CODE REVIEW

**Comment by saneax** [comment]
WIP, so please do not merge

**Comment by saneax** [comment]
please test this

**Comment by saneax** [comment]
please test this

**Comment by bodepd** [comment]
this patch conflates way too many changes. I would prefer if you can isolate the changes that only effect vagrant from those that potentially effect all builds (ie: things that touch userdata) and submit them separately.

File: build_scripts/make_userdata.sh [hunk 1]
```diff
@@ -38,7 +54,13 @@ then
        dpkg -i internal.deb
 fi
 apt-get update
-apt-get install -y puppet software-properties-common puppet-jiocloud jiocloud-ssl-certificate
+
+if [ "${cloud_provider}" == "hp"]; then
```
Comment (bodepd) [inline build_scripts/make_userdata.sh]: why would we only install certain packages in hp cloud and not for example when we move to our own cloud for testing? I would prefer if we limit the amount of per cloud differences as much as possible.

**Pull request closed without merging**